packages/material-ui-styles/src/ThemeProvider/ThemeProvider.test.js

```diff
@@ -53,6 +53,42 @@ describe('ThemeProvider', () => {
     assert.strictEqual(wrapper.text(), 'foobar');
   });
 
+  it('should memoize the merged output', () => {
```

Comment:
Just to be clear this test doesn't check if the child doesn't re-render which was part of the original issue. This only checks if the context value is properly memoized.

Reply:
It seems to be a React bug.

Reply:
The styles are injected when the theme reference changes. A rerender is not ideal but the styles are conserved.

Reply:
I don't understand why it rerender. But I can think of a better test. Hold on.

Reply:
Well then at least the actual issue is addressed. I want to investigate why this re-renders though. My intuition says it shouldn't.

Reply:
Ok, I'm keeping the test like this then. Hopefully, if it's a bug in react, we will see this test fail once it's fixed.